bot/bot_logic.py

```diff
@@ -0,0 +1,163 @@
+# Description: This file contains the logic for the LLM bot
+import os
+import openai
+import pandas as pd
+import numpy as np
+import json
+from typing import List, Optional
+import requests
+from dotenv import dotenv_values
+import psycopg2
+from psycopg2 import pool
+from psycopg2 import Error
+from psycopg2 import sql
+from langchain.docstore.document import Document
```

Comment:
This looks copied from the pgvector and langchain `pg_qna.py`. So, very specific rather than that can be adopted generically.